server/src/test/java/io/druid/client/CachingClusteredClientTest.java

```diff
@@ -2133,11 +2134,12 @@ public Result apply(
       DateTime timestamp = (DateTime) objects[index++];
 
       List<EventHolder> values = Lists.newArrayList();
+
       while (index < objects.length && !(objects[index] instanceof DateTime)) {
         values.add(new EventHolder(null, 0, (Map) objects[index++]));
       }
 
-      retVal.add(new Result<>(timestamp, new SelectResultValue(null, values)));
+      retVal.add(new Result<>(timestamp, new SelectResultValue(null, Sets.<String>newHashSet(), Sets.<String>newHashSet(), values)));
```

Comment:
can you add a caching test that has some actual values in dimensions and metrics set ?